doc/keynames/README.md

```diff
@@ -0,0 +1,213 @@
+# Key Names in Elektra
+
+## Preface
+
+This document is a full explanation of how Key Names work in Elektra.
```

Comment:
Why the capitalization? As we write "key-value", maybe we want "key-name"?

I did not read it now (I saw that there are still TODOs). Please tell me when it is ready to review.

Reply:
> Why the capitalization? As we write "key-value", maybe we want "key-name"?

The capitalization was very much intentional. It is a common sign of a proper noun in English. I used it to signify that e.g. "Key Name" means something very specific and is not just the name of a key. I actually capitalized all the terms a defined and I think I also italicized them in their definitions.

It also makes regex-replaces easy, if defined terms are capitalized, so I can replace it with "key-name" if you want to. Although in most of codebase we use "keyname", AFAIK. But my spell-checker really didn't like that (my hands don't like typing keyname either, so many typos....) so I avoided that.

> I did not read it now (I saw that there are still TODOs). Please tell me when it is ready to review.

As stated in the other comment, it is now ready for review.

Reply:
> It is a common sign of a proper noun in English.

Yes, proper nouns are capitalized but technical terms never get proper nouns (with the rare exceptions of one-of-a-kind items like the Internet). It is a very common thing to (re)define technical terms but this does not make them proper nouns.

In Elektra we define technical terms in the glossary doc/help/elektra-glossary.md and use them everywhere in this meaning but not capitalized.

Please also do so for key names (or key-names or keynames if you want, as long as it is everywhere the same). Otherwise it only brings inconsistency without any benefit.

>  spell-checker, typos

You can add your common typos in scripts/sed